sdk/identity/azure-identity/src/test/java/com/azure/identity/DefaultAzureCredentialTest.java

```diff
@@ -114,6 +117,35 @@ public void testUseAzureCliCredential() throws Exception {
         }
     }
 
+    @Test
+    public void testUseAzureDeveloperCliCredential() throws Exception {
+        // setup
+        String token1 = "token1";
+        TokenRequestContext request = new TokenRequestContext().addScopes("https://management.azure.com");
+        OffsetDateTime expiresAt = OffsetDateTime.now(ZoneOffset.UTC).plusHours(1);
+        EmptyEnvironmentConfigurationSource source = new EmptyEnvironmentConfigurationSource();
+        Configuration configuration = new ConfigurationBuilder(source, source, source).build();
+
+        // mock
+        try (MockedConstruction<IdentityClient> mocked = mockConstruction(IdentityClient.class, (identityClient, context) -> {
+            when(identityClient.authenticateWithAzureDeveloperCli(request)).thenReturn(TestUtils.getMockAccessToken(token1, expiresAt));
+            when(identityClient.authenticateWithAzureCli(request)).thenReturn(Mono.empty());
+            when(identityClient.authenticateWithManagedIdentityConfidentialClient(request)).thenReturn(Mono.empty());
+            when(identityClient.authenticateWithSharedTokenCache(request, null)).thenReturn(Mono.empty());
+            when(identityClient.authenticateWithIntelliJ(request)).thenReturn(Mono.empty());
+            when(identityClient.authenticateWithVsCodeCredential(any(), any())).thenReturn(Mono.empty());
+        }); MockedConstruction<IntelliJCredential> ijcredential = mockConstruction(IntelliJCredential.class, (intelliJCredential, context) -> {
+            when(intelliJCredential.getToken(request)).thenReturn(Mono.empty());
+        })) {
+
+            // test
+            DefaultAzureCredential credential = new DefaultAzureCredentialBuilder().configuration(configuration).build();
+            StepVerifier.create(credential.getToken(request)).expectNextMatches(accessToken -> token1.equals(accessToken.getToken()) && expiresAt.getSecond() == accessToken.getExpiresAt().getSecond()).verifyComplete();
```

Comment:
I'm not sure why this test (which I know you copied) only validates the second, but it should probably validate the whole timestamp or something? If I'm right about the timezone thing, it would have caught it here, so we should enhance this.